easybuild/easyblocks/v/vmd.py

```diff
@@ -0,0 +1,159 @@
+##
+"""
+EasyBuild support for VMD, implemented as an easyblock
+
+@author: Stephane Thiell (Stanford University)
+@author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+import shutil
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.easyblocks.generic.pythonpackage import det_pylibdir
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import change_dir
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root, get_software_version
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+
+
+class EB_VMD(ConfigureMake):
+    """Easyblock for building and installing VMD"""
+
+    def configure_step(self):
+        """
+        Configure VMD for building.
+        """
+        # make sure required dependencies are available
+        deps = {}
+        for dep in ['FLTK', 'Mesa', 'netCDF', 'Python', 'Tcl', 'Tk']:
+            deps[dep] = get_software_root(dep)
+            if deps[dep] is None:
+                raise EasyBuildError("Required dependency %s is missing", dep)
+
+        # optional dependencies
+        for dep in ['ACTC', 'CUDA', 'OptiX']:
+            deps[dep] = get_software_root(dep)
+
+        # specify Tcl/Tk locations & libraries
+        tclinc = os.path.join(deps['Tcl'], 'include')
+        tcllib = os.path.join(deps['Tcl'], 'lib')
+        env.setvar('TCL_INCLUDE_DIR', tclinc)
+        env.setvar('TCL_LIBRARY_DIR', tcllib)
+
+        env.setvar('TK_INCLUDE_DIR', os.path.join(deps['Tk'], 'include'))
+        env.setvar('TK_LIBRARY_DIR', os.path.join(deps['Tk'], 'lib'))
+
+        tclshortver = '.'.join(get_software_version('Tcl').split('.')[:2])
+        self.cfg.update('buildopts', 'TCLLDFLAGS="-ltcl%s"' % tclshortver)
+
+        # Python locations
+        pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+        env.setvar('PYTHON_INCLUDE_DIR', os.path.join(deps['Python'], 'include/python%s' % pyshortver))
+        pylibdir = det_pylibdir()
+        python_libdir = os.path.join(deps['Python'], os.path.dirname(pylibdir))
+        env.setvar('PYTHON_LIBRARY_DIR', python_libdir)
+
+        # numpy include location, easiest way to determine it is via numpy.get_include()
+        out, ec = run_cmd("python -c 'import numpy; print numpy.get_include()'", simple=False)
+        if ec:
+            raise EasyBuildError("Failed to determine numpy include directory: %s", out)
+        else:
+            env.setvar('NUMPY_INCLUDE_DIR', out.strip())
+
+        # compiler commands
+        self.cfg.update('buildopts', 'CC="%s"' % os.getenv('CC'))
+        self.cfg.update('buildopts', 'CCPP="%s"' % os.getenv('CXX'))
+
+        # source tarballs contains a 'plugins' and 'vmd-<version>' directory
+        vmddir = os.path.join(self.cfg['start_dir'], '%s-%s' % (self.name.lower(), self.version))
+
+        # first, build plugins
```

Comment:
well, I can provide a reference to the VMD docs, but that's how it should be done, see http://www.ks.uiuc.edu/Research/vmd/doxygen/compiling.html or http://robinbetz.com/blog/2015/01/08/compiling-vmd-with-python-support/